wts/resources/oauth2.py

```diff
@@ -18,19 +22,19 @@ def client_do_authorize():
     if mismatched_state:
         raise AuthError("could not authorize; state did not match across auth requests")
     try:
-        tokens = flask.current_app.oauth2_client.fetch_access_token(
-            redirect_uri, **flask.request.args.to_dict()
-        )
-        return refresh_refresh_token(tokens)
+        tokens = client.fetch_access_token(redirect_uri, **flask.request.args.to_dict())
+        return refresh_refresh_token(tokens, requested_idp)
     except KeyError as e:
         raise AuthError("error in token response: {}".format(tokens))
-    except (OAuth2Error, OAuthException) as e:
+    except AuthlibBaseError as e:
         raise AuthError(str(e))
 
 
-def find_valid_refresh_token(username):
+def find_valid_refresh_token(username, idp):
```

Comment:
Also think we should add test cases to `find_valid_refresh_token` now that the behavior has been changed to check for multiple idps.